tests/Behat/features/list-view.feature

```diff
@@ -0,0 +1,51 @@
+@javascript
+Feature: View a list of versions
+  As a cms author
+  I want to view a list past revisions of a versioned DataObject (incl. pages)
+
+  Background:
+    # Test date cannot be in the past or the version list numbers won't be descending
+    Given the current date is "2100-01-01"
+    Given a "page" "Home" with "Content"="Background"
+
+    Given I am logged in with "ADMIN" permissions
+    And I go to "/admin/pages"
+    And I click on "Home" in the tree
+
+  Scenario: A list of versions is displayed
+    Given I click on "History" in the header tabs
+    Then I should see a list of versions
+
+  Scenario: List shows the publish state, publish date and the author
+    Given I should see an edit page form
+    When I fill in the "Content" HTML field with "<p>Publish scenario</p>"
+    And I press the "Publish" button
+    Then I should see a "Published 'Home' successfully." notice
+    # This is a workaround @todo remove when https://github.com/silverstripe/silverstripe-cms/issues/2128 is resolved in framework
+    And I go to "/admin/pages/history/show/1"
+    Then I should see a list of versions
+    And I should see "ADMIN User" in the author column in version 1
+    And I should see "Published" in the record column in version 1
+    And I should see "01/01/2100" in the record column in version 1
+    And I should see the "Live" badge
+
+  Scenario: List shows the draft state, draft date and the author
+    Given I should see an edit page form
+    When I fill in the "Content" HTML field with "<p>Save scenario</p>"
+    And I press the "Save" button
+    Then I should see a "Saved 'Home' successfully." notice
+    # This is a workaround @todo remove when https://github.com/silverstripe/silverstripe-cms/issues/2128 is resolved in framework
+    And I go to "/admin/pages/history/show/1"
+    Then I should see a list of versions
+    And I should see "ADMIN User" in the author column in version 1
+    And I should see "Saved" in the record column in version 1
+    And I should see "01/01/2100" in the record column in version 1
```

Comment:
Maybe we should add `And I should not see the "Live" badge` to ensure it doesn't show up when nothing is published yet? We can modify the regex for the `And I should see the :arg badge` step definition to support an optional `not`

Reply:
I like the idea, but I am having trouble getting it to work at the moment. Below my approach:

```
/**
     * Example: I should see the "Live" badge
     * Example: I should not see the "Live" badge
     *
     * @Then /^I should( not? |\s*)see the :text badge
     * @param string $negative
     * @param string $text
     */
    public function iShouldSeeTheBadge($negative, $text)
    {
        if (trim($negative)) {
            $this->assertElementContains('.history-viewer .badge', $text);
        } else {
            $this->assertElementNotContains('.history-viewer .badge', $text);
        }
    }
```
How important do you reckon this is at this stage?

Reply:
I think you need to name the regex capture group for it to be assigned to a variable, e.g.:

```
@Then /^I should (?P<negative>(?:(not |)))see the :text badge
...
public function iShouldSeeTheBadge($negative, $text)
{
```